### ISSUE TITLE
New license request: 3D-Slicer-1.0

### ISSUE THREAD

**Issue by jcfr** [spdx/license-list-XML#2213]
**1.** License Name: 3D Slicer license

**2.** Short identifier: 3d-slicer-license

**3.** License Author or steward: Brigham and Women’s Hospital (BWH)

**4.** Comments:

> The 3D Slicer software is distributed under a BSD-style open source license that is broadly compatible with the Open Source Definition by The Open Source Initiative and contains no restrictions on legal uses of the software.
> 
> ### Historical notes about the license
> 
> See https://slicer.readthedocs.io/en/latest/user_guide/about.html#historical-notes-about-the-license
> 
> ### License terms and reasons
> 
> See https://slicer.readthedocs.io/en/latest/user_guide/about.html#license-terms-and-reasons
> 
> ### Status compared to other open source licenses
> 
> As of June 2021, the Slicer License has been used for over 15 years without incident. In May of 2021, a discourse user [suggested](https://discourse.slicer.org/t/apply-for-osi-open-source-license-status/17791) submitting the license to [the OSI license review process](https://opensource.org/approval). After some discussion and hearing no objections, the community leadership decided to [submit the license for review](https://lists.opensource.org/pipermail/license-review_lists.opensource.org/2021-May/thread.html). Although the OSI process is not legally binding, the discussion could give potential Slicer users perspective on how provisions of the license compare with other commonly used licenses. The discussion concluded that bundling the contribution agreement in the license makes it non-approvable by OSI and the requirement to use the software for legal purposes may not be consistent with the [Open Source Definition](https://opensource.org/osd). Otherwise the license terms appear not to be controversial. Interested parties should review the [full discussion](https://lists.opensource.org/pipermail/license-review_lists.opensource.org/2021-June/thread.html#5166) for details.
> 
> Copied from https://slicer.readthedocs.io/en/latest/user_guide/about.html#status-compared-to-other-open-source-licenses

**5.** License Request Url: http://tools.spdx.org/app/license_requests/312

**6.** URL(s):
* https://slicer.org/LICENSE
* https://github.com/Slicer/Slicer/blob/main/License.txt

**7.** OSI Status: Rejected

**8.** Example Projects:
* https://github.com/Slicer/Slicer
* https://github.com/Slicer/SlicerExecutionModel
* https://github.com/mhalle/spl-brain-atlas

**Comment by swinslow** [comment]
Discussed briefly on 2023-12-14 legal team call, @swinslow to review in detail and provide full analysis for consideration

**Comment by jcfr** [comment]
Thanks for follow-up :pray: 

In the meantime, let us know if you you have any questions.

cc: @lassoan @pieper @sjh26

**Comment by jlovejoy** [comment]
# New submission review

* **Has the license been approved by the OSI?** 
    - [ ] Yes
    - [X] No - it was submitted to OSI, some issues raised, and then submission was rescinded

## Definitive Factors

_These must all be satisfied to allow inclusion in the license list_

1. **Is the submitted license unique, that is, it does not match another license already on the License List as per the [matching guidelines](https://spdx.org/spdx-license-list/matching-guidelines)?**
    - [ ] Yes
    - [ ] No
1. **If a software license, does it apply to source code and not only to executables?**
    - [ ] Yes
    - [ ] No
1. **Does the license have identifiable and stable text, and is not in the midst of drafting?**
    - [ ] Yes
    - [ ] No
1. **Has the license steward, if any, committed to versioning new versions and to not modify it after addition to the list?**
    - [ ] Yes
    - [ ] No

## Other factors for inclusion

*Roughly in order of descending importance*

1. **Does the license substantially comply with one of the free/open content definitions? (examples include the [Open Source Definition](https://opensource.org/osd) and the [Debian Free Software Guidelines](https://www.debian.org/social_contract#guidelines))** 
    - the license part B seems to, but it also includes in part A a contributor license, which is simply unusual
1. **Is the license structured to be generally usable by anyone, and not specific to one organisation or project?**
    - [ ] Yes
    - [X] No - it is very much specific to the project and project maintainer
1. **Does the license have substantial use such that it is likely to be encountered (ie. use in many projects, or in one significant project)?** 
    - given it is specific to a project, functionality, and project maintainer, it won't be used or found in other projects. That being said, according to the Slicer website, it has been downloaded a million times, see https://slicer.readthedocs.io/en/latest/user_guide/about.html#historical-notes-about-the-license so we might consider that substantially use in and of itself?
1. **Is the license primarily intended to facilitate the free distribution of content with limited restrictions?**
    - Yes, but it is very, very specific to the code, type of code (has clauses related to medical software), and project maintainers
1. **Does the license steward support this submission, or is at least aware of and not in opposition of it?**
    - I believe so
## Summary of factors, outcome, comments

_Comments on the totality of these "other factors" in light of the SPDX License List's overall goals and objectives_

**Comment by pieper** [comment]
Thank you for the review.  Let us know if there are any specifics we can help resolve.

**Comment by karsten-klein** [comment]
**ScanCode** 
Identifier: 3dslicer-1.0
Links
* [Link to ScanCode](https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/3dslicer-1.0.LICENSE)

**metaeffekt universe** 
short identifier: 3D-Slicer-1.0 (we preserve original case)
canonical name: 3D Slicer Contribution and Software License Agreement 1.0
* [Link to universe](https://github.com/org-metaeffekt/metaeffekt-universe/blob/main/src/main/resources/ae-universe/[3]/README.md#:~:text=3D%20Slicer%20Contribution%20and%20Software%20License%20Agreement%201.0)
* [Link to metadata](https://github.com/org-metaeffekt/metaeffekt-universe/blob/main/src/main/resources/ae-universe/%5B3%5D/%5B3d%5D/3D-Slicer-Contribution-and-Software-License-Agreement-1.0.yaml)

Other hints:
I would not include the "-license" on the short name. This is until now done very rarely (some exceptional BSD cases). The predominant convention - as it appears to me - is not to include a "-license" suffix on shortId.

**Comment by jcfr** [comment]
Thanks for taking the time to comment and provide references :pray: 

Being consistent with existing best practices and conventions makes sense. I definitely support the proposed naming :ok_hand: 

-----

Explicitly specifying `1.0` is indeed sensible considering the version is explicitly mentioned in the license text :100: 

<img width=50% src="https://github.com/spdx/license-list-XML/assets/219043/6f5cead3-d9d2-4c94-bb06-689a52cd49ba"/>

**Comment by jlovejoy** [comment]
Thanks @karsten-klein  for the input on naming!

Do you have any insight as to the "substantial use" factor? (see my comment above)

**Comment by karsten-klein** [comment]
I'm afraid, I don't have any substantial information on the use of the license.

**Comment by jcfr** [comment]
> Do you have any insight as to the "substantial use" factor?

Over the past decade, the project has been downloaded more than 1.5 million times[^1], with over 270,000 downloads in just the past year alone. Moreover, Slicer was used in more than 17,000 peer reviewed publications[^2]. Beyond its academic and research applications, the software also serves as the foundation for various commercial applications, including FDA-approved ones[^3].

It may also be worth noting that the license was developed by Harvard Medical School in the context of collaborative NIH-funded research involving 15 institutions, including MIT, GE, UNC, and other commercial and academic organizations. All parties involved carefully reviewed and signed off on the license. Subsequent reviews by other organizations have supported ongoing growth of the community.

Thanks again for your time reviewing our submission :pray:.

In the meantime, let us know if you have any additional questions. Also, if you think that would be helpful, we have a weekly[^4] public video hangout with user and developer of the medical imaging community and would be happy to further discuss.

_Reply co-authored by @pieper, @lassoan and @jcfr in the context of the Slicer weekly community meeting of March 26th._

[^1]: https://download.slicer.org/download-stats/
[^2]: https://scholar.google.com/scholar?hl=en&as_sdt=1%2C22&as_vis=1&q=%28%223D+Slicer%22+OR+%22slicer+software%22+OR+%22slicer+org%22+OR+Slicer3D%29+-Slic3r+&btnG=
[^3]: https://www.slicer.org/commercial-use.html
[^4]: Weekly at 10am ET, announced at https://discourse.slicer.org/c/community/hangout/20

**Comment by swinslow** [comment]
Thank you all for your comments (and my apologies for not getting to the writeup -- thanks very much @jlovejoy for the analysis above).

Given @jlovejoy's review, and the details from @jcfr regarding usage, I'm also +1 to add this to the list. Per @karsten-klein's note I agree that it shouldn't include '-license' as the suffix to the identifier. Discussed on 2024-05-09 legal team call and agreed to add with `3D-Slicer-1.0` as the identifier, given the 1.0 version number in the license text.

**Comment by swinslow** [comment]
# License Inclusion Decision

## Decision:

- [X] approved
- [ ] not approved

## Name

3D Slicer License

## License ID

`3D-Slicer-1.0`

## XML markup

None

## Notes

Likely none, will confirm during XML submission

## Next steps ##

@swinslow to add if no one else does prior to the 3.24 release

**Comment by swinslow** [comment]
@jcfr Purely as a heads-up, in the course of putting together the PR to add this (#2476) I noticed that there are a couple of typo differences between the license text at https://github.com/Slicer/Slicer/blob/main/License.txt, and the version from the Slicer website at https://slicer.org/LICENSE:

1. In Part A.6, the GitHub version has the word "REQUIRES" where the Slicer website version has "REQURES"
2. In Part B.1, the GitHub version has the word "Women's" where the Slicer website version has "Women?s". (I checked the bytes and it does actually appear to be a question mark / hex $3F, not some other weird Unicode character being rendered strangely.)

In #2476 I've included "alt text" regexes which should match to either of these, for SPDX matching purposes. But just mentioning in case the Slicer project decides that it wants to address these typos in the LICENSE version at https://slicer.org/LICENSE  :)

Let me know if you have any questions, thanks!
